algorithm.py

```diff
@@ -20,11 +24,8 @@
 
 Tutorial
 --------
-
-This module provides an abstraction for implementing arbitrary algorithms as a
-list of functions that operate on a shared state dictionary. Algorithms defined
-this way are easy to arbitrarily modify at run time, and they provide cascading
-exception handling.
+The algorithm model is a list of functions that are executed in
+defined order and update algorithm state stored in `state` dictionary.
```

Comment:
This sounds better, so I updated the wording. It also raises the question if algorithm can be non-linear? Does it support branching and parallel lines like real flow chart?

https://en.wikipedia.org/wiki/Algorithm#/media/File:Euclid_flowchart.svg